src/PactNet.Native/NativeMessagePactBuilder.cs

```diff
@@ -0,0 +1,123 @@
+    {
+        private readonly IMessageMockServer server;
+        private readonly MessagePactHandle pact;
+        private readonly PactConfig config;
+        private MessageHandle message;
+
+        /// <summary>
+        /// Initialises a new instance of the <see cref="NativeMessagePactBuilder"/> class.
+        /// </summary>
+        /// <param name="server">Mock server</param>
+        /// <param name="pact">the messagePact handle</param>
+        /// <param name="config">the messagePact configuration</param>
+        internal NativeMessagePactBuilder(IMessageMockServer server, MessagePactHandle pact, PactConfig config)
+        {
+            this.pact = pact;
+            this.config = config ?? throw new ArgumentNullException(nameof(config));
+            this.server = server ?? throw new ArgumentNullException(nameof(server));
+        }
+
+        #region IMessagePactBuilderV3 explicit implementation
+
+        /// <inheritdoc cref="IMessagePactBuilderV3"/>
+        IMessageBuilderV3 IMessagePactBuilderV3.ExpectsToReceive(string description)
+            => ExpectsToReceive(description);
+
+        /// <inheritdoc cref="IMessagePactBuilderV3"/>
+        IMessagePactBuilderV3 IMessagePactBuilderV3.WithPactMetadata(string @namespace, string name, string value)
+            => WithPactMetadata(@namespace, name, value);
+
+        #endregion
+
+        #region Internal Methods
+
+        /// <summary>
+        /// Add a new message to the messagePact
+        /// </summary>
+        /// <param name="description">Message description</param>
+        /// <returns>Fluent builder</returns>
+        internal IMessageBuilderV3 ExpectsToReceive(string description)
+        {
+            this.message = this.server.NewMessage(pact, "default message");
+
+            this.server.ExpectsToReceive(this.message, description);
+
+            return new NativeMessageBuilder(this.server, this.message);
+        }
+
+        /// <summary>
+        /// Add a new metadata to the messagePact
+        /// </summary>
+        /// <param name="namespace">the parent configuration section</param>
+        /// <param name="name">the metadata field value</param>
+        /// <param name="value">the metadata field value</param>
+        /// <returns>Fluent builder</returns>
+        internal IMessagePactBuilderV3 WithPactMetadata(string @namespace, string name, string value)
+        {
+            this.server.WithMessagePactMetadata(this.pact, @namespace, name, value);
+
+            return this;
+        }
+
+        /// <summary>
+        /// Verify a message is read and handled correctly and write the messagePact
+        /// </summary>
+        /// <param name="handler">The method using the message</param>
+        public void Verify<T>(Action<T> handler)
+        {
+            try
+            {
+                var content = JsonConvert.DeserializeObject<NativeMessage>(this.server.Reify(message));
```

Comment:
I'm a bit confused by this bit - why do message pacts take metadata, content types, headers, etc. but then we only ever use the message body? What uses those extra fields?